### PR TITLE
Round Sprinty McBurndown's output to integers

### LINKED ISSUE #1850: [Task]: Round Sprinty McBurndown's post to nearest integer
### Summary

Currently the stats included in the Sprinty McBurndown is rounded to the 100th decimal place. We instead want to round to the nearest whole percentage point. See screenshot below:

Screenshot 2024-04-26 at 11 19 37 AM

### Acceptance criteria

Stats included in the Sprinty McBurndown posts are rounded to the nearest whole percentage point.

### PR DESCRIPTION
**Pull request by AlexanderStephensonUSDS**
## Summary
Fixes #1850 

## Changes proposed

Adds code to metrics to covert float percentages to whole integers.

## Additional information

See Screenshot from local CLI.
burndown
burnup
percent_deliver

### CODE REVIEW

**Comment by coilysiren** [comment]
@AlexanderStephensonUSDS let me know if you need help with the CI failures

**Comment by AlexanderStephensonUSDS** [comment]
@coilysiren What do I need to fix for the Anchore Scan vulnerability check failure?

**Comment by coilysiren** [comment]
@AlexanderStephensonUSDS if you look at the check, you'll see this

image

It's referencing the current version of `Jinja2`, which is being installed here: https://github.com/HHS/simpler-grants-gov/blob/779201c4c6c5d95a81cd60245d1103d785629c3f/analytics/poetry.lock#L899-L900

To fix the scan, you need to update `Jinja2`. You can do that with the following command:

```bash
$ poetry update
```

**Review by coilysiren — DISMISSED** [review]

File: analytics/src/analytics/metrics/burndown.py [hunk 1]
```diff
@@ -105,12 +105,12 @@ def get_stats(self) -> dict[str, Statistic]:
         # get open and closed counts and percentages
         total_opened = int(df["opened"].sum())
         total_closed = int(df["closed"].sum())
-        pct_closed = round(total_closed / total_opened * 100, 2)
+        pct_closed = int(total_closed / total_opened * 100)
```
Comment (widal001) [inline analytics/src/analytics/metrics/burndown.py:108]: I think we might actually want this: 
```suggestion
        pct_closed = int(round(total_closed / total_opened * 100, 0))
```

`int()` by itself just floors a float and we want to round to the nearest whole number

File: analytics/tests/metrics/test_burndown.py [hunk 2]
```diff
@@ -356,7 +356,7 @@ def test_include_issues_closed_after_sprint_end(self):
         # validation
         assert output.stats.get(self.TOTAL_CLOSED).value == 2
         assert output.stats.get(self.TOTAL_OPENED).value == 3
-        assert output.stats.get(self.PCT_CLOSED).value == 66.67  # rounded to 2 places
+        assert output.stats.get(self.PCT_CLOSED).value == 66  # rounded to 2 places
```
Comment (widal001) [inline analytics/tests/metrics/test_burndown.py:359]:
```suggestion
        assert output.stats.get(self.PCT_CLOSED).value == 67  # rounded to nearest whole number
```

Changed this sounds it reflects the rounding to the nearest whole number

**Review by widal001 — CHANGES_REQUESTED** [review]
Hey @AlexanderStephensonUSDS thanks for finding and making these changes in the necessary places!

One small edit (which I should have made more clear in the ticket) we want to round the numbers to the nearest whole percentage point rather than simply flooring them to their integer value.

I left a few code suggestions that you should be able to accept, but can you update the other uses of `int()` alone as well?

**Comment by AlexanderStephensonUSDS** [comment]
Yes. Makes sense.

Will make the requested changes.

On May 10, 2024, at 10:49 AM, Billy Daly ***@***.***> wrote:

﻿

@widal001 requested changes on this pull request.

Hey @AlexanderStephensonUSDS<https://github.com/AlexanderStephensonUSDS> thanks for finding and making these changes in the necessary places!

One small edit (which I should have made more clear in the ticket) we want to round the numbers to the nearest whole percentage point rather than simply flooring them to their integer value.

I left a few code suggestions that you should be able to accept, but can you update the other uses of int() alone as well?

________________________________

In analytics/src/analytics/metrics/burndown.py<https://github.com/HHS/simpler-grants-gov/pull/1976#discussion_r1596916888>:

> @@ -105,12 +105,12 @@ def get_stats(self) -> dict[str, Statistic]:
         # get open and closed counts and percentages
         total_opened = int(df["opened"].sum())
         total_closed = int(df["closed"].sum())
-        pct_closed = round(total_closed / total_opened * 100, 2)
+        pct_closed = int(total_closed / total_opened * 100)


I think we might actually want this:

⬇️ Suggested change

-        pct_closed = int(total_closed / total_opened * 100)
+        pct_closed = int(round(total_closed / total_opened * 100, 0))


int() by itself just floors a float and we want to round to the nearest whole number

________________________________

In analytics/tests/metrics/test_burndown.py<https://github.com/HHS/simpler-grants-gov/pull/1976#discussion_r1596919569>:

> @@ -356,7 +356,7 @@ def test_include_issues_closed_after_sprint_end(self):
         # validation
         assert output.stats.get(self.TOTAL_CLOSED).value == 2
         assert output.stats.get(self.TOTAL_OPENED).value == 3
-        assert output.stats.get(self.PCT_CLOSED).value == 66.67  # rounded to 2 places
+        assert output.stats.get(self.PCT_CLOSED).value == 66  # rounded to 2 places


⬇️ Suggested change

-        assert output.stats.get(self.PCT_CLOSED).value == 66  # rounded to 2 places
+        assert output.stats.get(self.PCT_CLOSED).value == 67  # rounded to nearest whole number


Changed this sounds it reflects the rounding to the nearest whole number

**Review by widal001 — DISMISSED** [review]
LGTM! 

I ran `make sprint-reports-with-latest-data ACTION=post-results` and got the following results:

- [Sprint burndown](https://betagrantsgov.slack.com/archives/C06L7JGT9M0/p1715609506850519)
- [Percent complete by deliverable](https://betagrantsgov.slack.com/archives/C06L7JGT9M0/p1715609515827489)

**Comment by widal001** [comment]
We'll tackle this in one of the other related issues in #2355

**Pull request closed without merging**